frontend/src/component/Account/PersonalData.jsx

```diff
@@ -24,6 +23,13 @@ export class PersonalData extends React.Component {
     }
   }
 
+  isFieldWritable = field => {
+    return !this.props.displayAdminInfo &&
+      this.props.system.config.user__read_only_fields &&
+      this.props.system.config.user__read_only_fields[this.props.userAuthType] &&
+      this.props.system.config.user__read_only_fields[this.props.userAuthType].includes(field)
+  }
+
```

Comment:
it is more usual to name this kind of function "isEditable"
also, you could take advantage of [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) to reduce the "&&"

And lastly, I think you are using "props.displayAdminInfo" as a "isAdmin" kind of props. I would prefer not to.
I think, since the information about "isAdmin" comes from props, it is the parent component that should give a props for this specific usage. Or, simpler way, give a isAdmin props and update displayAdminInfo props

Reply:
Noted for the name and the operator (since I saw it first and only in TS, I thought this was exclusive to Typescript), will use it now !

From the usage of it, I seemed that `props.displayAdminInfo` is not really a `isAdmin`, but is only set to `true` if the user is an admin. Perhaps renaming this prop then ?

Reply:
I think renaming the props is a good solution

About the optional chaining, I don't think we have already use it in tracim and it's a recent addition in js.
So you must check that it is properly transpilled by frontend build. If not, it's better not to use it before upgrading Babel

Reply:
I decided to make it more readable, without the operator. I think we already had this discussion, and the answer was to not use it

Reply:
I think too, sadly :<